### PR TITLE
[TECHDEBT][Persistence] Fix Persistence Module TODOs - Consolidate Actor (issue-#210)

### LINKED ISSUE #210: [TECHDEBT][Persistence] Fix Persistence Module TODOs - Consolidate Actor
# Objective
A general cleanup issue is needed to tackle TODO's and ensure the persistence module is usable/readable by consolidating `Actor` with `BaseActor` as part of https://github.com/pokt-network/pocket/issues/172. 

# Origin Document
Should follow issue-#128, issue-#105, issue-#147 and issue-#148 the persistence module is messier and more difficult to understand than the developers would want for organic external contribution.

# Goals / Deliverables
- [ ] Consolidate `Actor` in `persistence/schema/base_actor.go` with `BaseActor` in `persistence/schema/base_actor.go`.

## General issue checklist
- [ ] Update the appropriate CHANGELOG
- [ ] Update the README
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)
- [ ] Update any relevant global documentation & references
- [ ] Document small issues / TODOs along the way
---

Creator: @andrewnguyen22
Co-creator: @Olshansk

### PR DESCRIPTION
**Pull request by andrewnguyen22**
## Description

A general cleanup issue is needed to tackle TODO's and ensure the persistence module is usable/readable by consolidating `Actor` with `BaseActor` as part of https://github.com/pokt-network/pocket/issues/172. 

Follows issue-#128, issue-#105, issue-#147 and issue-#148 the persistence module is messier and more difficult to understand than the developers would want for organic external contribution.

## Issue

Fixes #210 

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- [x] Consolidate `Actor` in `persistence/schema/base_actor.go` with `BaseActor` in `persistence/schema/base_actor.go`.

## Testing

- [x] `make develop_test`
- [x] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Required Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested my changes using the available tooling
- [x] I have updated the corresponding CHANGELOG

### If Applicable Checklist
- [x] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: persistence/test/setup_test.go [hunk 1]
```diff
@@ -161,9 +161,9 @@ func fuzzSingleProtocolActor(
 		switch op {
 		case "UpdateActor":
 			numParamUpdatesSupported := 3
-			newStakedTokens := originalActor.StakedTokens
+			newStakedTokens := originalActor.StakedAmount
 			newChains := originalActor.Chains
-			newActorSpecificParam := originalActor.ActorSpecificParam
+			newActorSpecificParam := originalActor.GenericParam
```
Comment (Olshansk) [inline persistence/test/setup_test.go:166]: Not a blocker to merge this in, but just starting a thread for a quick convo

We use `ActorSpecificParam` or `GenericParam` to represent `MaxRelays`, `ServiceURI`, etc... We had https://github.com/pokt-network/pocket/issues/206 originally to do some consolidation, but are you okay with us openining another ticket to improve this in the future? 

Doesn't even have to be an M1 tasks, but maybe more M3 or M4 related once things are more mature.

**Review by Olshansk — APPROVED** [review]
Nice and simple. Ty!

Left one comment just to have a short convo, but no blockers from merging in.

**Side note**: What do you think about `FisherPeople` or `FishFolks` instead of `Fisherman` long-term?

File: persistence/test/setup_test.go (hunk 1 above)
Comment (andrewnguyen22) [reply persistence/test/setup_test.go:166]: Yes - I noticed this as well. I think we need another consolidation / improve ticket. 

Are you thinking strictly consolidation, if so I can open the tix - else I might need your help to craft it

Comment (Olshansk) [reply persistence/test/setup_test.go:166]: Crafted here: #313

**Pull request merged**